zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/cluster/TestSpringApplication.java

```diff
@@ -190,6 +192,18 @@ public final T withAuthenticatedAccess() {
     return withUnauthenticatedAccess(false);
   }
 
+  /**
+   * Sets the broker's working directory, aka its data directory. If a path is given, the broker
+   * will not delete it on shutdown.
+   *
+   * @param directory path to the broker's root data directory
+   * @return itself for chaining
+   */
+  public final T withWorkingDirectory(final Path directory) {
```

Comment:
:wrench: A working directory is only for brokers, so the proper fix would be to extract an interface for applications which use the broker role.